stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt

```diff
@@ -320,10 +332,20 @@ public class MessageComposerController(
     /**
      * Sets up the observing operations for various composer states.
      */
+    @OptIn(FlowPreview::class)
     private fun setupComposerState() {
         input.onEach { input ->
             state.value = state.value.copy(inputValue = input)
-        }.launchIn(scope)
+
+            if (canSendTypingUpdates.value) {
+                typingUpdatesBuffer.onKeystroke()
+            }
+            handleCommandSuggestions()
+            handleValidationErrors()
+        }.debounce(ComputeMentionSuggestionsDebounceTime)
+            .onEach {
+                handleMentionSuggestions()
```

Comment:
Since we debounce handling mentions by 300ms, there's a slight delay between clicking on a mention and ind appearing in the input. I don't perceive this delay much, but please keep it in mind when testing and tell me if it is noticeable. 

If it is we'll try debouncing only network searches, however that will raise the complexity.

Reply:
Slightly noticeable, not a dealbreaker.

We could experiment pushing the debounce down a bit, e.g. 250ms or 200, but as-is is okay too